logger-cli/src/logger.rs

```diff
@@ -0,0 +1,156 @@
+// shared-core - bitdrift's common client/server libraries
+// Copyright Bitdrift, Inc. All rights reserved.
+//
+// Use of this source code is governed by a source available license that can be found in the
+// LICENSE file or at:
+// https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt
+
+use crate::cli::{FieldPairs, LogCommand, Options};
+use crate::metadata::Metadata;
+use crate::storage::SQLiteStorage;
+use bd_client_common::error::handle_unexpected;
+use bd_logger::{CaptureSession, InitParams, Logger};
+use bd_session::{Strategy, fixed};
+use bd_test_helpers::metadata_provider::LogMetadata;
+use parking_lot::Mutex;
+use std::path::{Path, PathBuf};
+use std::pin::Pin;
+use std::sync::Arc;
+
+pub type LoggerFuture =
+  Pin<Box<dyn Future<Output = anyhow::Result<()>> + 'static + std::marker::Send>>;
+
+pub const SESSION_FILE: &str = "session_id";
+
+#[allow(dead_code)]
+pub struct LoggerHolder {
+  logger: Logger,
+  future: Mutex<Option<LoggerFuture>>,
+  shutdown_trigger: bd_shutdown::ComponentShutdownTrigger,
+}
+
+impl LoggerHolder {
+  pub fn new(
+    logger: Logger,
+    future: LoggerFuture,
+    shutdown_trigger: bd_shutdown::ComponentShutdownTrigger,
+  ) -> Self {
+    Self {
+      logger,
+      future: Mutex::new(Some(future)),
+      shutdown_trigger,
+    }
+  }
+
+  pub fn start(&self) {
+    let Some(future) = self.future.lock().take() else {
+      return;
+    };
+
+    sleep(2000); // some initialization time for the network
+
+    std::thread::spawn(move || {
+      tokio::runtime::Builder::new_current_thread()
+        .enable_all()
+        .build()
+        .unwrap()
+        .block_on(async {
+          handle_unexpected(future.await, "logger top level run loop");
+        });
+    });
+  }
+
+  pub fn process_crash_reports(&mut self) -> anyhow::Result<()> {
+    self
+      .logger
+      .process_crash_reports(bd_logger::ReportProcessingSession::PreviousRun)
+  }
+
+  pub fn stop(&self) {
+    sleep(2000);
+    self.logger.shutdown(true);
+  }
+
+  pub fn log(&self, cmd: LogCommand, capture_session: bool) {
+    let session_capture = if capture_session {
+      CaptureSession::capture_with_id("cli command")
+    } else {
+      CaptureSession::default()
+    };
+    self.logger.new_logger_handle().log(
+      cmd.log_level.into(),
+      cmd.log_type.into(),
+      cmd.message.into(),
+      FieldPairs(cmd.field).into(),
+      [].into(),
+      None,
+      bd_logger::Block::Yes,
+      session_capture,
+    );
+  }
+}
+
+fn sleep(milliseconds: u64) {
+  std::thread::sleep(std::time::Duration::from_millis(milliseconds));
+}
```

Comment:
a neat pattern here would be something like

```
fn sleep(duration: std::time::Duration) {
  std::thread::sleep(duration)
}
```

then you can call this with `sleep(200.std_milliseconds())` 

using the time::ext::NumericalStdExtension trait